docs/reference/mistakecheckers.md

```diff
@@ -0,0 +1,16 @@
+## Mistake Checkers
+This refers to validating user actions against what is expected and considered correct in a lab.
+## Types of Errors
+In the simulation, there are a variety of possible things to check for, though they tend to resolve to two categories: general and lab-specific. 
+### General
+These may include configuring the `CurrentSource` wires incorrectly. This is something that a `LabObject` should be aware of and let the user know of their mistake.
+### Lab-Specific
+These include issues that are relevant to a specific lab. For instance, the run time of the gel in the Gel Electrophoresis module should not be something a `LabObject` like the `CurrentSource` or `ElectrolysisSetup` should be aware of.
+## Implementation
+The general structure relies on the Strategy Design Pattern where each separate `MistakeChecker` type implements the base `MistakeChecker` class.
+### `MistakeChecker` functions
+- `CheckAction(params: Dictionary)`: checks the action based off the parameters sent by `ReportAction()`. It should compare the `actionType` of `params` to determine whether it should do anything. It then should send a `LabLog` log, warn, or error depending on the requirements.
```

Comment:
I think we should maybe give an example of this here - maybe take an existing `actionType` and list what other parameters it should have?

Reply:
Good idea! I added an example for checking heat time for the gel.